protobuf_definitions/message_formats.proto

```diff
@@ -0,0 +1,319 @@
+/**
+ * Common messages
+ *
+ * These are used for logging as well as building requests and responses.
+ */
+
+syntax = "proto3";
+
+package blueye.protocol;
+option csharp_namespace = "Blueye.Protocol.Protobuf";
+
+/**
+ * Motion input from client.
+ *
+ * Used to indicate the desired motion in each direction.
+ * Typically these values map to the left and right joystick for motion,
+ * and the left and right trigger for the slow and boost modifiers.
+ */
+message MotionInput {
+  float surge = 1; // Forward (positive) and backwards (negative) movement. (-1..1).
+  float sway = 2; // Right (positive) and left (negative) lateral movement (-1..1).
+  float heave = 3; // Descend (positive) and ascend (negative) movement (-1..1).
+  float yaw = 4; // Left (positive) and right (negative) movement (-1..1).
+  float slow = 5; // Multiplier used to reduce the speed of the motion (0..1).
+  float boost = 6; // Multiplier used to increase the speed of the motion (0..1).
+}
+
+message UserModifiers {
+  float surge_modifier = 1;
+  float sway_modifier = 2;
+  float heave_modifier = 3;
+  float yaw_modifier = 4;
+  float jerk_modifier = 5;
+}
+
+message Lights {
+  uint32 percentage = 1;
+}
+
+message LatLongPosition {
+  float latitude = 1;
+  float longitude = 2;
+}
+
+message ConnectionDuration {
+  int64 connection_duration = 1;
+}
+
+message AutoHeadingState {
+  bool enabled = 1;
+}
+
+message AutoDepthState {
+  bool enabled = 1;
+}
+
+message SystemTime {
+  string unix_timestamp = 1;
```

Comment:
Why is this a string? To deal with the 2038-issue?

Reply:
@sindrehan Not sure. I just left it from your initial commit. Haven't implemented it yet. It's a uint32 for number of seconds since epoch in current protocol.

Reply:
But uin32 should be ok for 2038 right, so maybe we should go back to that?

Reply:
Or we can use the standard timestamp message: https://developers.google.com/protocol-buffers/docs/reference/google.protobuf#google.protobuf.Timestamp

Reply:
I used the google.protobuf.Timestamp Message: 03cda01d762df81d3fadf534da4a67c68eed3b51